pkg/apis/infinispan/v1/types_util.go

```diff
@@ -313,23 +319,50 @@ func (ispn *Infinispan) IsEncryptionCertSourceDefined() bool {
 	return ee != nil && ee.Type != ""
 }
 
-type StatusUpdateFn func()
-type StatusValidateFn func() bool
+type UpdateFn func()
+type ValidateFn func() bool
+
+func (ispn *Infinispan) Update(client clnt.Client, logger logr.Logger, validate ValidateFn, update UpdateFn) error {
+	infinispan := &Infinispan{}
+	err := client.Get(context.TODO(), types.NamespacedName{Namespace: ispn.Namespace, Name: ispn.Name}, infinispan)
+	if err != nil {
+		logger.Error(err, "failed to reload Infinispan")
```

Comment:
No need to log errors in the `Update` method as we always return the `err` instance in the infinispan_controller if it's not null. Logging the error twice just adds verbosity

Reply:
The same applies to the `UpdateStatus` method.

Reply:
Yeah, agree with you @ryanemerson 
But the problem describe in #676 says, that second error trace (from the controller-runtime) fully informal and don't display stack trace and lace of the error
I would prefer keep this "as is" and discuss and implement this at the your issue #690

Reply:
> I would prefer keep this "as is" and discuss and implement this at the your issue #690

Makes sense. We can ensure we're consistent with our approach that way.